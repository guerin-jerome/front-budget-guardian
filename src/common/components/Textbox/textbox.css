input {
  height: 20px;
  width: 100%;
  padding: 8px;
  margin-top: 5px;
  font-size: 14.5px;
  font-family: Arial, sans-serif;
  border-radius: 2px;
  border: 1px solid #ccc;
}
