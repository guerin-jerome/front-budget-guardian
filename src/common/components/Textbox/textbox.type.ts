import { ChangeEventHandler } from "react";
import { UseFormRegisterReturn } from "react-hook-form";

export type TextboxProps = Partial<{
  id: string;
  register: UseFormRegisterReturn;
  type: "text" | "number" | "date";
  placeholder: string;
  step: number;
  testId: string;
  hasError: boolean;
  onChange: ChangeEventHandler<HTMLInputElement>;
}>;
