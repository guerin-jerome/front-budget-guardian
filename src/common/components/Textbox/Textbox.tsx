import { TextboxProps } from "./textbox.type";
import "./textbox.css";

export const Textbox = ({
  step,
  placeholder,
  type = "text",
  register,
  testId,
}: TextboxProps) => (
  <input
    data-testid={testId}
    type={type}
    step={step}
    placeholder={placeholder}
    {...register}
  />
);
