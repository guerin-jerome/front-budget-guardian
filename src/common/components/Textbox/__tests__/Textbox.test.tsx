import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { Textbox } from "../Textbox";
import userEvent from "@testing-library/user-event";

describe("<Textbox />", () => {
  it("matchs snapshot", () => {
    const { container } = render(<Textbox />);
    expect(container).toMatchSnapshot();
  });

  it("renders well - type number", async () => {
    render(<Textbox type="number" step={2} placeholder="Ex: 100" />);
    const textbox = screen.getByRole("spinbutton");
    await userEvent.type(textbox, "100");
    expect(textbox).toBeDefined();
  });
});
