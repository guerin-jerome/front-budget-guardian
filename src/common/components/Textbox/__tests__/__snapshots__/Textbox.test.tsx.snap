// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Textbox /> > matchs snapshot 1`] = `
<div>
  <input
    type="text"
  />
</div>
`;
